src/modules/components/transaction/transactionDataListItem/transactionDataListItemStructure/transactionDataListItemStructure.api.ts

```diff
@@ -0,0 +1,71 @@
+import { type Hash } from 'viem';
+import { IconType, type AvatarIconVariant, type IDataListItemProps } from '../../../../../core';
+
+export enum TxStatusCode {
+    PENDING = 'PENDING',
+    SUCCESS = 'SUCCESS',
+    FAILED = 'FAILED',
+}
+
+export enum TransactionType {
+    DEPOSIT = 'DEPOSIT',
+    WITHDRAW = 'WITHDRAW',
+    ACTION = 'ACTION',
+}
+
+export const txHeadingStringList: Record<TransactionType, string> = {
+    [TransactionType.DEPOSIT]: 'Deposit',
+    [TransactionType.WITHDRAW]: 'Withdraw',
+    [TransactionType.ACTION]: 'Smart contract action',
+};
+
+export const txIconTypeList: Record<TransactionType, IconType> = {
+    [TransactionType.DEPOSIT]: IconType.DEPOSIT,
+    [TransactionType.WITHDRAW]: IconType.WITHDRAW,
+    [TransactionType.ACTION]: IconType.BLOCKCHAIN_SMARTCONTRACT,
+};
+
+export const txVariantList: Record<TransactionType, AvatarIconVariant> = {
+    [TransactionType.DEPOSIT]: 'success',
+    [TransactionType.WITHDRAW]: 'warning',
+    [TransactionType.ACTION]: 'info',
+};
+
+export interface ITransactionDataListItemProps extends IDataListItemProps {
+    /**
+     * The chain ID of the transaction.
+     */
+    chainId: number;
+    /**
+     * The address of the token.
+     */
+    tokenAddress?: string;
+    /**
+     * The symbol of the token, e.g. 'ETH' as a string
+     */
+    tokenSymbol?: string;
+    /**
+     * The token value in the transaction.
+     */
+    tokenValue?: number;
+    /**
+     * The type of transaction.
+     */
+    txType: TransactionType;
+    /**
+     * The current status of a blockchain transaction on the network.
+     */
+    txStatus: TxStatusCode;
+    /**
+     * The Unix timestamp of the transaction.
+     */
+    formattedTimestamp?: string;
+    /**
+     * The estimated fiat value of the transaction.
+     */
+    fiatEstimate?: number;
```

Comment:
On the `assetDataListItem` component we ended up calculating the transaction value based on the token-price and token-amount. I would follow the same approach here, by adding a `tokenPrice` property and calculate the fiatAmount as `tokenAmount * tokenPrice`. 
@sepehr2github Handled these two properties to accept both `number | string`. I think it would be nice to do this here as well as sometimes we might get these data as strings or number